### DISCUSSION TITLE
Excessive IO usage

### DISCUSSION THREAD

**Discussion by YeapGuy** [nextcloud/all-in-one#1272] [Questions] [answered]
Hi,
After trying to diagnose random disk spinups, I came across a tool called `fatrace`, which outputs all file accesses on the entire system to the terminal. When I ran it, my terminal started getting literally spammed by it, and when I stopped it, the file access logs were all pretty much like this:
```
grep(2472649): O /mnt/dietpi_userdata/docker-data/volumes/nextcloud_aio_mastercontainer/_data/session/sess_57b09a32ab12b5969241d7c81d0dcd94
grep(2472649): RC /mnt/dietpi_userdata/docker-data/volumes/nextcloud_aio_mastercontainer/_data/session/sess_57b09a32ab12b5969241d7c81d0dcd94
grep(2472650): O /mnt/dietpi_userdata/docker-data/volumes/nextcloud_aio_mastercontainer/_data/session/sess_3cf578d0abedcf7c06693e414342dc63
grep(2472650): RC /mnt/dietpi_userdata/docker-data/volumes/nextcloud_aio_mastercontainer/_data/session/sess_3cf578d0abedcf7c06693e414342dc63
grep(2472651): O /mnt/dietpi_userdata/docker-data/volumes/nextcloud_aio_mastercontainer/_data/session/sess_63d4f1ee6548d8fed90de09a830f4067
grep(2472651): RC /mnt/dietpi_userdata/docker-data/volumes/nextcloud_aio_mastercontainer/_data/session/sess_63d4f1ee6548d8fed90de09a830f4067
grep(2472652): O /mnt/dietpi_userdata/docker-data/volumes/nextcloud_aio_mastercontainer/_data/session/sess_e57a28bfce85f7ec28ec3fb88965f1c0
grep(2472652): RC /mnt/dietpi_userdata/docker-data/volumes/nextcloud_aio_mastercontainer/_data/session/sess_e57a28bfce85f7ec28ec3fb88965f1c0
grep(2472653): O /mnt/dietpi_userdata/docker-data/volumes/nextcloud_aio_mastercontainer/_data/session/sess_11c499ba5502b56118e93df3a3ba3419
grep(2472653): RC /mnt/dietpi_userdata/docker-data/volumes/nextcloud_aio_mastercontainer/_data/session/sess_11c499ba5502b56118e93df3a3ba3419
```
What the heck is this??? Why are the PHP sessions getting accessed like 10 times each second?! I run Nextcloud-AIO on a Raspberry Pi with an SD card, this is going to wear it out insanely fast!
Now, my first question is why is this even happening, and how can I prevent it?
If no one can answer that, would it be fine if I mounted the PHP session directory to RAM so I don't wreck my SD card in a month?

**Comment by szaimen** [comment]
Please calm down!

First of all AIO is not meant to be run on an SD-card. So your specific setup is not supported. 

Second, should reading files not burn out your SD-card but only writing, IIRC. 

Third, I would honestly suggest moving the docker directory onto a SSD for much better performance. See https://github.com/nextcloud/all-in-one#how-to-store-the-filesinstallation-on-a-separate-drive

Fourth, this should in theory work, indeed:
> would it be fine if I mounted the PHP session directory to RAM so I don't wreck my SD card in a month?

**Reply by YeapGuy** [reply]
Even if AIO "isn't meant to be run on an SD card", I don't understand why the PHP session files need to be read so often. Sure, I'll mount it to RAM, but I'm still curious why it does this?

**Reply by szaimen** [reply]
Indeed you have a point here. It will get improve with https://github.com/nextcloud/all-in-one/pull/1274

**Reply by YeapGuy** [reply]
Excellent, thanks for the fix! I was just about to also mention that as another issue: there's WAY too many session files in there. 😅

**Reply by YeapGuy** [reply]
@szaimen Right, I tried patching my mastercontainer with that PR, but I don't think the behavior changed? How can I verify this?

**Reply by szaimen** [reply]
I suppose you didn't do the patching correctly. please wait for the next release.

**Reply by YeapGuy** [reply]
Okay, I'll wait. Mounted the session directory to RAM for now. Thanks for your quick assistance and fix implementation, looking forward to the next version 👀

**Comment by szaimen** [comment, marked as answer]
This is now fixed with v2.1.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
